### ISSUE TITLE
Issue printing to Matlab console

### ISSUE THREAD

**Issue by nikhar-abbas** [OpenFAST/openfast#156]
I have been working with @rafmudaf to get simulink connected to openfast. The latest commit on [my run_simulink branch](https://github.com/nikhar-abbas/openfast/tree/run_simulink) has a working connection between the simulink and openfast interfaces, but openfast is not currently printing to the Matlab console. 

[modules-local/nwtc-library/src/SysMatlab_Unix.f90](https://github.com/nikhar-abbas/openfast/blob/run_simulink/modules-local/nwtc-library/src/SysMatlab_Unix.f90) has been created to attempt to solve this issue, but I am having some trouble with getting mexPrintf to work. Any advice here?

**Comment by bjonkman** [comment]
Can you explain this problem a bit more? I'm not exactly sure what "having trouble getting mexPrintf to work" means.

Does it not link when you build the OpenFAST library (I don't see anything in the cmake files that specifically looks for the Matlab `mex` library it needs to link with)? Are you running a Simulink model? Is it printing to some file?

**Comment by nikhar-abbas** [comment]
In some previous changes, a flag was added to `~/openfast/CMakeLists.txt` to include a CMake build option to set the COMPILE_SIMULINK flag found in `ServoDyn.f90`. These updates, along with some minor changes to `FAST_SFunc.c` and `create_FAST_SFunc.m` to account for the FAST.farm additions and to point to the right install directories made it possible to run fast via Simulink, but without Fast printing to the Matlab command window. In this case, Simulink is not printing to a file, just to an output structure in the Matlab workspace.

In order to account for this, an option was added to `~/openfast/modules-local/nwtc-library/CMakeLists.txt` to set the NWTC_SYS_FILE as `src/SysMatlab_Unix.f90` if Fast is being compiled for use with simulink. 

As I had previously stated, I am having trouble with setting up `SysMatlab_Unix.f90` to print to the command window. You are correct, my trouble is getting cmake to link with Matlab. You had commented at the beginning of SysMatlab.f90 
`   ! bjj: if compiling this to write to the Matlab command window using mexPrintf, you  must link with libmex.lib in the matlab/extern/lib/{architecture}/{compiler} folder otherwise, use preprocessor definition CONSOLE_FILE to output everything to a file named CONSOLE.TXT  `

but I am struggling with realizing how to achieve this on Mac OSX with Matlab R2018a. Some Mathworks documentation and googling suggested using `#include "fintrf.h"`, but this is both inconsistent with the older SysMatlab.f90 practice, and I cannot seem to get it to compile still. 

Hope this clears things up a bit, thanks!

**Comment by bjonkman** [comment]
Yes, I saw the `COMPILE_SIMULINK` flag added to the cmake files, but I don't see where it adds the mex library (libmex) file to the linking command. In cmake, this would probably be in a way similar to how it adds the mkl libraries to the linking command. Maybe @rafmudaf could help you set that up.

I don't see anything that would have to change in the SysMatlab_Unix.f90 file, but if you are still having issues, it would be helpful if I could read the actual error messages.

** Also, you said your changes "made it possible to run fast via Simulink, but without Fast printing to the Matlab command window. In this case, Simulink is not printing to a file, just to an output structure in the Matlab workspace." Are you sure the screen output isn't in a `fort.6` file (assuming `CU=6` in your Sys*.f90 file) in your Matlab working directory?

**Comment by nikhar-abbas** [comment]
We have added `~/openfast/cmake/FindMatlab.cmake` to automate the search for the matlab directories (this may need to be cleaned up a little bit). In doing so, and adding a `find_package(Matlab REQUIRED)` statement to `~/openfast/CMakeLists.txt', it seems like the libraries have been linked. We are still struggling with succesfuly calling mexprintf in SysMatlab_Unix.f90, though. 

The original error message was :
```
Undefined symbols for architecture x86_64:
  "_mexprintf_", referenced from:
      _syssubs_mp_wrnr_ in SysMatlab_Unix.f90.o
      _syssubs_mp_wrover_ in SysMatlab_Unix.f90.o
      _syssubs_mp_writescr_ in SysMatlab_Unix.f90.o
ld: symbol(s) not found for architecture x86_64
make[2]: *** [modules-local/nwtc-library/libnwtclibs.dylib] Error 1
make[1]: *** [modules-local/nwtc-library/CMakeFiles/nwtclibs.dir/all] Error 2
make: *** [all] Error 2
```

As suggested by https://www.mathworks.com/help/matlab/apiref/mexprintf.html, we tried adding `#include "fintrf.h"` at the beginning of the subroutines calling mexprintf.  The`fintrf.h` header file exists in `/Applications/MATLAB_R2018a.app/extern/include`, and we then get the following error message when compiling:
```
SysMatlab_Unix.f90(398): #error: can't find include file: fintrf.h
```

As far as the `fort.6` file, that does not seem to exist in the Matlab working directory. `CU=6` in the SysIFL.f90 file that was used to compile everything succesfully.

**Comment by bjonkman** [comment]
Those error messages still sound like linker didn't use the mex library.  You've got some code to find the Matlab libraries, but I don't see it getting used in `target_link_libraries` in `modules-local/nwtc-library/CMakeLists.txt`. 

By the way, that "fintrf.h" file is just a bunch of `#define` statements, so I wouldn't include it twice in the code. I'd be surprised if you even needed it at all.

**Comment by nikhar-abbas** [comment]
Thanks a ton for the help @bjonkman and @rafmudaf ! It looks like adding/changing 
```
add_library(nwtclibs ${NWTCLIBS_SOURCES})
if(BUILD_OPENFAST_SIMULINK_API)
  find_package(Matlab REQUIRED)
  target_link_libraries(nwtclibs ${LAPACK_LIBRARIES} ${CMAKE_DL_LIBS} ${Matlab_MEX_LIBRARY})
else()
  target_link_libraries(nwtclibs ${LAPACK_LIBRARIES} ${CMAKE_DL_LIBS})
endif()
```
in `/openfast/modules-local/nwtc-library/CMakeLists.txt` resolved the issue. No changes were necessary in `/openfast/CMakeLists.txt` as far as linking the matlab libraries is concerned. 

The openfast output is now printing to the matlab command window, albeit without time updates (though I don't recall that ever being the case, and is easily printed via simulink).

After some minor code cleanup, these changes will be pushed to https://github.com/nikhar-abbas/openfast/tree/run_simulink.

**Comment by rafmudaf** [comment]
Thanks @bjonkman and @nikhar-abbas. Once this issue is fixed, it will also close #59.

**Comment by bjonkman** [comment]
This code was merged into OpenFAST/dev, so I think this can be closed, @rafmudaf.
